source/_posts/2019-04-23-release-v12.0.0.md

```diff
@@ -0,0 +1,731 @@
+    * `.getColorDepth()`로 색상을 확인합니다. (Vladislav Kaminsky) [#26261](https://github.com/nodejs/node/pull/26261)
+    * REPLServer.rli는 폐기 예정입니다. (Ruben Bridgewater) [#26260](https://github.com/nodejs/node/pull/26260)
+* **src**:
+    * 사용하지 않는 `INT_MAX` 상수를 제거했습니다. (Sam Roberts) [#27078](https://github.com/nodejs/node/pull/27078)
+    * `NODE_MODULE_VERSION`을 72로 업데이트했습니다. (Ujjwal Sharma) [#26685](https://github.com/nodejs/node/pull/26685)
+    * `AddPromiseHook()`을 제거했습니다. (Anna Henningsen) [#26574](https://github.com/nodejs/node/pull/26574)
+    * `MultiIsolatePlatform` 인터페이스를 정리했습니다. (Anna Henningsen) [#26384](https://github.com/nodejs/node/pull/26384)
+    * 기본 힙 제한을 적절하게 설정합니다. (Ali Ijaz Sheikh) [#25576](https://github.com/nodejs/node/pull/25576)
+    * node 설정에서 `icuDataDir`를 제거했습니다. (GauthamBanasandra) [#24780](https://github.com/nodejs/node/pull/24780)
+* **tls**:
+    * TLSv1.3을 지원합니다. (Sam Roberts) [#26209](https://github.com/nodejs/node/pull/26209)
+    * `getCipher()`에서 올바른 버전을 반환합니다. (Sam Roberts) [#26625](https://github.com/nodejs/node/pull/26625)
+    * renegotiate()에서 인자의 타입을 검사합니다.  (Sam Roberts) [#25876](https://github.com/nodejs/node/pull/25876)
+    * `ERR_TLS_INVALID_PROTOCOL_METHOD`의 코드를 추가했습니다. (Sam Roberts) [#24729](https://github.com/nodejs/node/pull/24729)
+    * 서버명이 IP 주소일 때 경고합니다. (Rodger Combs) [#23329](https://github.com/nodejs/node/pull/23329)
+    * 기본적으로 TLS v1.0과 v1.1를 비활성화합니다. (Ben Noordhuis) [#23814](https://github.com/nodejs/node/pull/23814)
+    * createSecureContext()에서 사용하지 않는 인자를 제거했습니다. (Sam Roberts) [#24241](https://github.com/nodejs/node/pull/24241)
+    * `Server.prototype.setOptions()`는 폐기 예정입니다. (cjihrig) [#23820](https://github.com/nodejs/node/pull/23820)
+    * 시작할 때 `NODE_EXTRA_CA_CERTS`를 로드합니다. (Ouyang Yadong) [#23354](https://github.com/nodejs/node/pull/23354)
+- * **util**:
+    * `util.print()`, `util.puts()`, `util.debug()`, `util.error()`를 제거했습니다. (cjihrig) [#25377](https://github.com/nodejs/node/pull/25377)
+    * inspect의 compact와 breakLength의 기본값을 변경했습니다. (Ruben Bridgewater) [#27109](https://github.com/nodejs/node/pull/27109)
+    * inspect의 엣지 케이스를 개선했습니다. (Ruben Bridgewater) [#27109](https://github.com/nodejs/node/pull/27109)
+    * 순환 참조 오류에서 오류 메시지의 첫 줄만 비교합니다. (Simon Zünd) [#26685](https://github.com/nodejs/node/pull/26685)
+    * callbackify로 만들어진 함수의 프로토타입을 설정하지 않습니다. (Ruben Bridgewater) [#26893](https://github.com/nodejs/node/pull/26893)
+    * callbackify로 만들어진 함수의 이름을 변경합니다. (Ruben Bridgewater) [#26893](https://github.com/nodejs/node/pull/26893)
+    * `callbackify()`를 사용할 때 함수의 길이를 증가시켰습니다. (Ruben Bridgewater) [#26893](https://github.com/nodejs/node/pull/26893)
+    * `inspect()`의 내부에서 조작되는 것을 막았습니다. (Ruben Bridgewater) [#26577](https://github.com/nodejs/node/pull/26577)
+    * `.inspect()`가 Proxy 트랩을 일으키지 않도록 막았습니다.  (Ruben Bridgewater) [#26241](https://github.com/nodejs/node/pull/26241)
```

Comment:
원문은 `prevent Proxy traps being triggered by .inspect()`인데 여기서 traps의 의미가 무엇일까요?

Reply:
unexpected trigger를 그렇게 적은거 아닐까요.

Reply:
PR을 보니 `getPrototypeOf`, `setPrototypeOf`, `isExtensible` 등을 직접 설정할 수 있고, 그게 `.inspect()` 도중에 호출되었던 것 같네요. https://github.com/nodejs/node/pull/26241/files 테스트 쪽 코드를 보시면 `handler`에 전부 Error 던지는 것들로 채워넣고 Proxy를 만든 다음 `.inspect()`를 호출해서 오류 없이 실행되는 걸 확인하고 있습니다.

PR의 맨 윗부분을 보면 이렇게 써 있습니다.

> // Always check for proxies to prevent side effects and to prevent triggering
> // any proxy handlers.
> const proxy = getProxyDetails(value);